File: Beginner/PlusOne/plusOne.cpp
#include <vector>
#include <iostream>

using namespace std;

class Solution {
    public: 
        std::vector<int> plusOne(std::vector<int>& digits) {
            for(int i = digits.size() -1; i >= 0; --i) {
                digits[i]++;
                digits[i] %= 10;

                if (digits[i] != 0) {
                    return digits;
                }
            }

            digits.insert(digits.begin(), 1);

            return digits;
        }
};

#include <iostream> // Include the necessary header file

int main() {
    Solution s;
    std::vector<int> digits = {1, 2, 3};
    std::vector<int> result = s.plusOne(digits);
    for (int i = 0; i  < result.size(); i++) {
        std::cout << result[i] << " ";
    }
    return 0;
}

File: Beginner/PlusOne/plusOne.py
from typing import List


class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        num_digits = len(digits)

        for i in range(num_digits -1, -1, -1):
            digits[i] += 1

            digits[i] %= 10

            if digits[i] != 0:
                return digits
        return [1] + digits

s = Solution()
print(s.plusOne([1,2,3])) # [1,2,4]
print(s.plusOne([9])) # [1,0]
print(s.plusOne([9,9])) # [1,0]
print(s.plusOne([9,8]))
